system-test/kitchen.ts

```diff
@@ -0,0 +1,189 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as assert from 'assert';
+import {describe, it, beforeEach} from 'mocha';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as tmp from 'tmp';
+import * as crypto from 'crypto';
+import * as os from 'os';
+import {Readable} from 'stream';
+import {createURI, ErrorWithCode, upload} from '../src/gcs-resumable-upload';
+
+const bucketName = process.env.BUCKET_NAME || 'gcs-resumable-upload-test';
+tmp.setGracefulCleanup();
+const tmpFileContents = crypto.randomBytes(1024 * 1024 * 20);
+const filePath = path.join(os.tmpdir(), '20MB.zip');
+fs.createWriteStream(filePath).write(tmpFileContents);
```

Comment:
This stream should be closed to avoid open handle issues (https://nodejs.org/api/fs.html#filehandlecreatereadstreamoptions)

Perhaps write stream would be better easier? https://nodejs.org/api/fs.html#fswritesyncfd-buffer-offset-length-position